### PR TITLE
Five segments stable for testing

### PR DESCRIPTION
**Pull request by jr1221**
## Changes

_Explanation of changes goes here_

Combination of five segment fixups and compute 24a branch into the mega car-worky branch.

## Notes

_Any other notes go here_

## Test Cases

- Case A
- Edge case
- ...

## To Do

_Any remaining things that need to get done_

- [ ] item 1
- [ ] ...

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please reach out to your Project Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #)

### CODE REVIEW

File: Core/Src/analyzer.c [hunk 1]
```diff
@@ -174,6 +174,7 @@ void calc_cell_voltages(acc_data_t *bmsdata)
 		uint8_t num_cells = get_num_cells(&bmsdata->chip_data[chip]);
 
 		for (uint8_t cell = 0; cell < num_cells; cell++) {
+			int current_state = 0;
```
Comment (caiodasilva2005) [inline Core/Src/analyzer.c]: current state will never be charging state like this

**Review by caiodasilva2005 — CHANGES_REQUESTED** [review]
Just one small thing. Thanks for helping me out with compute/segment testing stuff :)
